src/libraries/Common/src/System/Security/Cryptography/SlhDsa.cs

```diff
@@ -1114,109 +1320,160 @@ protected virtual void Dispose(bool disposing)
 
         private AsnWriter ExportSubjectPublicKeyInfoCore()
         {
-            byte[] rented = CryptoPool.Rent(Algorithm.PublicKeySizeInBytes);
+            // Public key size for SLH-DSA is at most 64 bytes so we can stack allocate it.
+            int publicKeySizeInBytes = Algorithm.PublicKeySizeInBytes;
+            Debug.Assert(publicKeySizeInBytes is <= 64 and >= 0);
+            Span<byte> buffer = stackalloc byte[publicKeySizeInBytes];
 
-            try
-            {
-                Span<byte> keySpan = rented.AsSpan(0, Algorithm.PublicKeySizeInBytes);
-                ExportSlhDsaPublicKey(keySpan);
+            ExportSlhDsaPublicKeyCore(buffer);
 
-                AsnWriter writer = new AsnWriter(AsnEncodingRules.DER);
+            // The ASN.1 overhead of a SubjectPublicKeyInfo encoding a public key is 18 bytes.
+            // Round it off to 32. This checked operation should never throw because the inputs are not
+            // user provided.
+            int capacity = checked(32 + publicKeySizeInBytes);
+            AsnWriter writer = new AsnWriter(AsnEncodingRules.DER, capacity);
```

Comment:
@bartonjs this would be another good example of where having value-type ASN.1 structs would be helpful. I don't like using the hand-cobbled ASN.1 encoding of things, but on the other hand we would lose out on the `stackalloc` here.

No real action needs to be taken here... other than maybe one day we will get those value type codecs.

Reply:
Yep, I had to revert to this, but I was also trying to find a way to use the `stackalloc` with the SPKIAsn struct. I'm curious how can this be solved with new APIs though? I think the main restriction is that Span/ReadOnlySpan can't be stored as a field.

Actually, if `SubjectPublicKeyInfoAsn` was generic on the `SubjectPublicKey` field then we can pass an inline array in maybe?

Reply:
@bartonjs has a prototype where `SubjectPublicKeyInfoAsn` has an equivalent `SubjectPublicKeyInfoValueAsn` (or whatever it will be called) that is a `ref struct`, so it can have members that are also `ref struct`. The hard part is how those types are generated.